experiments/react-native-uifabric/src/components/RadioGroup/RadioButton.tsx

```diff
@@ -0,0 +1,115 @@
+/** @jsx withSlots */
+'use strict';
+import * as React from 'react';
+import { View, Text, NativeSyntheticEvent } from 'react-native';
+import {
+  radioButtonName,
+  IRadioButtonType,
+  IRadioButtonProps,
+  IRadioButtonState,
+  IRadioButtonSlotProps,
+  IRadioButtonRenderData
+} from './RadioButton.types';
+import { compose, IUseComposeStyling } from '@uifabricshared/foundation-compose';
+import { filterViewProps } from '../../utilities/RenderHelpers';
+import { ISlots, withSlots } from '@uifabricshared/foundation-composable';
+import { settings } from './RadioButton.settings';
+import { mergeSettings } from '@uifabricshared/foundation-settings';
+import { foregroundColorTokens, textTokens, borderTokens, backgroundColorTokens } from '../../tokens';
+import { useAsPressable } from '../../hooks';
+import { RadioGroupContext } from './RadioGroup';
+
+export const RadioButton = compose<IRadioButtonType>({
+  displayName: radioButtonName,
+
+  usePrepareProps: (userProps: IRadioButtonProps, useStyling: IUseComposeStyling<IRadioButtonType>) => {
+    const { content, buttonKey, disabled, ariaLabel, ...rest } = userProps;
+
+    // Grabs the context information from RadioGroup (currently selected button and client's onChange callback)
+    const info = React.useContext(RadioGroupContext);
+
+    const pressable = useAsPressable(rest);
+
+    const state: IRadioButtonState = {
+      info: {
+        ...pressable.state,
+        selected: info.selectedKey === userProps.buttonKey,
+        disabled: disabled || false
+      }
+    };
+
+    // Grab the styling information from the userProps, referencing the state as well as the props.
+    const styleProps = useStyling(userProps, (override: string) => state.info[override] || userProps[override]);
+
+    // This function is called every time a RadioButton gains focus. It does two things:
+    // 1) Calls pressable's onFocus in order to keep track of our state's focus variable. It is dependent on pressable's
+    //    focus variable. Without this, it wouldn't stay updated because we're overriding it's onFocus below for the rootProps.
+    // 2) Selects the currently focused button by calling the RadioGroup's callback function.
+    const onFocusChange = React.useCallback(
+      (ev: NativeSyntheticEvent<{}>) => {
+        pressable.props.onFocus && pressable.props.onFocus(ev);
+        // This check is necessary because this func gets called even when a button loses focus (not sure why?) which then calls the client's onChange multiple times
+        if (!state.info.selected) {
```

Comment:
instead of checking for selected state, try

if (pressable.props.onFocus && pressable.props.onFocus(ev)) 
{
  info.onChange(buttonKey)
}

Reply:
wait, actually why don't you check for the focused state in pressable.state